include/veriblock/pop/blockchain/base_block_tree.hpp

```diff
@@ -519,6 +519,15 @@ struct BaseBlockTree {
     tips_.insert(index);
   }
 
+  //! dangerous! permanently drops the block from the tree
+  void eraseBlockUnchecked(index_t& block) {
+    VBK_LOG_INFO("Deallocating block %s", block.toShortPrettyString());
+    auto count = blocks_.erase(makePrevHash(block.getHash()));
+    VBK_ASSERT_MSG(count == 1,
+                   "state corruption: block %s is not in the store",
+                   block.toPrettyString());
+  }
+
   //! dangerous! permanently drops the block from the tree
   void eraseBlock(index_t& block) {
```

Comment:
I am not convinced that Unsafe func is needed. It doesn't check that pnext is empty, which means that we can deallocate a block in a middle of a chain. 
I'd remove Unsafe version.

Reply:
This is a refactoring commit. so, all it is intended to do is deduplicate code and regularize logging. 
Whether this function has a future depends on what happens to eraseBlock. We either want eraseLeaf+eraseRoot combo or a generalized eraseBlock that ensures we can only delete a root or a leaf. eraseLeaf/Root will have a use for eraseBlockUnchecked and I personally lean towards this approach as it is more explicit